submissions/Kostiantyn_Rakhmanov/oop-exercise/index.js

```diff
@@ -0,0 +1,65 @@
+import { print } from './js/lib.js';
+
+class Inhabitant {
+   constructor(species, name, gender, saying, legs) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.legs = legs;
+   }
+   printOut() {
+      return ['species', 'name', 'gender', 'saying', 'legs',].map(
+         (prop) => `${prop}: ${this[prop]}`
+      ).join('; ');
+   }
+}
+
+class Animal extends Inhabitant {
+   constructor(species, name, gender, legs, saying) {
+      super(species, name, gender, legs, saying);
+   }
+}
+
+class Dog extends Animal {
+   constructor(name, gender, saying, legs = 4) {
+      super('dog', name, gender, saying, legs);
+   }
+}
+
+class Cat extends Animal {
+   constructor(name, gender, saying, legs = 4) {
+      super('cat', name, gender, saying, legs);
```

Comment:
Explain what `super` is. It is not a rhetoric question.
This code contains two bugs. They annihilate each other. This is called "transposition of errors".
Can you spot it?
Please note that "it works" is not a good excuse, as bugs tend to reveal themselves during the further development of the project.

Reply:
Hi! Super consist of all that goes to Parent class.